File: 5-kyu/Perimeter of Squares in Rectangle/PerimeterOfSquaresInRectangle.py
# Kata located at: https://www.codewars.com/kata/559a28007caad2ac4e000083

def perimeter(n):
	i = 2
	prev = 1
	curr = 1
	res = 1 if n == 0 else 2
	while i <= n:
		curr += prev
		prev = curr - prev
		res += curr
		i += 1
	return res * 4
    

File: 5-kyu/Perimeter of Squares in Rectangle/PerimeterOfSquaresInRectangle.c
// Kata located at: https://www.codewars.com/kata/559a28007caad2ac4e000083

typedef unsigned long long ull;
ull perimeter(int n) {

 		ull prev = 1;
    ull curr = 1;
    ull answr = (n == 0) ? 1 : 2;
    
    for (int i=2; i<=n; ++i) {
    
      curr = prev + curr;
      prev = curr - prev;
      answr += curr;
      
    }
    return answr*4;
}
